docs/content/guides/developer/app-examples/reviews-rating.mdx

```diff
@@ -0,0 +1,205 @@
+
+## How reviews are rated
+
+The reviews are rated (scored) on-chain using following criteria
+- Intrinsic score (IS)
+  - Length of content
+- Extrinsic score (ES)
+  - Number of votes received
+- Verification multiplier (VM)
+  - Reviews with PoE get a multiplier to improve rating
+
+```
+Total Score = (IS + ES) * VM
+```
+
+## Smart contracts
+
+There are several modules that create the backend logic for the example.
+
+### dashboard.move
+
+The `dashboard.move` module defines the `Dashboard` struct that is used to group services.
+
+```move
+struct Dashboard has key, store {
+  id: UID,
+  service_type: String
+}
+```
+
+The services are grouped by attributes, which can be cuisine type, geographical location, operating hours, google maps ID, and so on. To keep it basic, the example stores only `service_type` (for example, fast food, Chinese, Italian).
+
+### service.move
+
+This module defines the `Service` struct, which service owners manage.
+
+```move
+struct Service has key, store {
+  id: UID,
+  reward_pool: Balance<SUI>,
+  reward: u64,
+  top_reviews: MultiMap<ID>,
+  reviews: Table<ID, ID>,
+  moderators: Table<address, address>,
+  overall_rate: u64,
+  name: String
+}
+```
+
+All the submitted reviews are stored in `reviews` field, and top rated reviews are stored in `top_reviews` field, which has `MultiMap<ID>` type. The [multimap module](#multimap) defines the type.
+
+The `reward_pool` and `reward` fields store the available reward information.
+
+```move
+struct AdminCap has key, store {
+  id: UID,
+  service_id: ID
+}
+struct Moderator has key {
+  id: UID,
+  service_id: ID,
+}
+```
+`SERVICE OWNERS` are given `AdminCap`, and `MODERATORS` are given `Moderator` to allow them to perform privileged operations.
+
+### multimap.move {#multimap}
+
+`MultiMap` is an implementation of the `multimap` data structure (https://en.wikipedia.org/wiki/Multimap), which keeps all the reviews sorted. Using `MultiMap`, all the reviews are ordered by rating, with the highest rated reviews coming first.
+
+```move
+struct MultiMap<K: copy> has copy, drop, store {
+  contents: vector<Entry<K>>,
+}
+struct Entry<K: copy> has copy, drop, store {
+  key: K,
+  priority: u64,
+}
+```
+The `object_id` of a `review` object is used as a `key` and `total_score` is used as `priority` for a new entry.
+
+Note that `vector` is used to store the contents of the `MultiMap`. This is acceptable in our use case because the rewards are distributed to 10 participants at most.
+
+```move
+public fun insert<K: copy>(self: &mut MultiMap<K>, key: K, priority: u64) {
+  assert!(!contains(self, &key), EKeyAlreadyExists);
+  let len = vector::length(&self.contents);
+  if (len == 0) {
+    vector::push_back(&mut self.contents, Entry { key, priority });
```

Comment:
This special case should be unnecessary, I think (you can insert at `0` in an empty vector).

Reply:
Yes, it is indeed unnecessary check. I've removed the check.